### DISCUSSION TITLE
Installation Error

### DISCUSSION THREAD

**Discussion by patelbhavinoutlook** [nocodb/nocodb#3864] [Q&A]
Hello,

I am new to NocoDB so may be this issue would sound very primary but this is show stopper for me to start using nocodb.

i have installed noco db with MySQL as per document.

docker run -d --name nocodb-mysql \
-v "$(pwd)"/nocodb:/usr/app/data/ \
-p 8080:8080 \
-e NC_DB="mysql2://host.docker.internal:3306?u=root&p=mypassword&d=testdb" \
-e NC_AUTH_JWT_SECRET="569a1821-0a93-45e8-87ab-eb857f20a010" \
nocodb/nocodb:latest

installation gets completed no issues. but when I tried using nocodb on HTTP://myip:8080 I get below error...

Cannot GET /

don't know what has gone wrong.. 

i have installed MySQL on same machine and I tried login with root also and its working fine.

**Comment by o1lab** [comment]
You would have to provide more details where you are running the docker. And whether the port is exposed etc.

Moving this to discussion.

**Reply by patelbhavinoutlook** [reply]
OS : Ubuntu 20.04,
Mysql: installed locally
Server: local hardware server
Database name: testdb

**Reply by patelbhavinoutlook** [reply]
Sever IP: 192.168.18.138
after installation accessing link 
HTTP://192.168.18.38:8080

**Reply by patelbhavinoutlook** [reply]
Mysql installed locally..

**Reply by o1lab** [reply]
Are you sure that port is open ?

**Reply by patelbhavinoutlook** [reply]
i did some other trials now .. 
some changes I did with a number of systems. This time I took two systems both ubuntu 20.04. The first system already has a MySQL server running(192.168.1.135). second also with ubuntu 20.04 (192.168.1.138). From the second system, I tried to connect MySQL on the command line and it was successful I was able to connect MySQL server on system 1 from system 2 with the user name and password.

Installed nocodb for testing (on second system with IP 192.168.1.138) with one command install as per the video guide with the below command line.

docker run --name artnocodb -p 8080:8080 -d nocodb/nocodb

Installation was successful, and I could log in on the port 808 web interface. from there i tried to connect the MySQL database to other systems with IP 192.168.1.135. It was successful. i was able to create a table and delete tables in the database from the web interface.

After this, i tried to do other installations (on system 192.168.1.138) as below which was given in a guide for installation of MySQL server. 

docker run -d --name nocodb-mysql \
-v "$(pwd)"/nocodb:/usr/app/data/ \
-p 8080:8080 \
-e NC_DB="mysql2://192.168.1.135:3306?u=testdb&p=password&d=testdb" \
-e NC_AUTH_JWT_SECRET="569a1821-0a93-45e8-87ab-eb857f20a010" \
nocodb/nocodb:latest

But this time it fails.. after installation when i tried to access the web interface it gave the below error which i got previous.

Cannot GET /

**Reply by patelbhavinoutlook** [reply]
Hello

**Reply by patelbhavinoutlook** [reply]
> Are you sure that port is open ?

==============================
i did some other trials now ..
some changes I did with a number of systems. This time I took two systems both ubuntu 20.04. The first system already has a MySQL server running(192.168.1.135). second also with ubuntu 20.04 (192.168.1.138). From the second system, I tried to connect MySQL on the command line and it was successful I was able to connect MySQL server on system 1 from system 2 with the user name and password.

Installed nocodb for testing (on second system with IP 192.168.1.138) with one command install as per the video guide with the below command line.

docker run --name artnocodb -p 8080:8080 -d nocodb/nocodb

Installation was successful, and I could log in on the port 808 web interface. from there i tried to connect the MySQL database to other systems with IP 192.168.1.135. It was successful. i was able to create a table and delete tables in the database from the web interface.

After this, i tried to do other installations (on system 192.168.1.138) as below which was given in a guide for installation of MySQL server.

docker run -d --name nocodb-mysql
-v "$(pwd)"/nocodb:/usr/app/data/
-p 8080:8080
-e NC_DB="mysql2://192.168.1.135:3306?u=testdb&p=password&d=testdb"
-e NC_AUTH_JWT_SECRET="569a1821-0a93-45e8-87ab-eb857f20a010"
nocodb/nocodb:latest

But this time it fails.. after installation when i tried to access the web interface it gave the below error which i got previous.

Cannot GET /

==========================================

**Reply by patelbhavinoutlook** [reply]
[Hello,]([url](url))

Can any one please help me to understand what issue is going on... why just simple installation as per guide not working for me. I am not doing any thing else.. even server is fresh Ubuntu 20.04 installed.. 

Also, i did one more testing.. i have installed MySQL on the same server and installed Nocodb with one command.

docker run --name artnocodb -p 8080:8080 -d nocodb/nocodb

I can access web interface on 8080 port even can work with SQL lite data base. 

But 

Can not connect MySQL server on localhost which is installed in same machine on which i installed noncodb on docker.

https://snipboard.io/34qgrA.jpg

**Reply by wingkwong** [reply]
Have you tried `host.docker.internal` instead of `localhost`?

**Reply by patelbhavinoutlook** [reply]
I get error as below


https://paste.pics/IXDRP

**Comment by patelbhavinoutlook** [comment]
I got something worked.. actually, Mysql was installed on the same server as the services.. so i was able to access MySQL database below as

MySQL -u testdb -p (and giving password). 

also can access Mysql with IP of server 192.168.18.138

I check IP of the docker container which was created with single command docker creation.. IP was 172.17.0.2, I added rule to open port 3306 in the firewall for origin IP 172.17.0.2. 

now on server IP that is 192.168.18.138.. i can connect MySQL server database from the interface.. but for localhost, it gives some error

![image](https://user-images.githubusercontent.com/28344691/193536381-3c0db547-7c5e-4924-b1ab-4da9296b403a.png)


For host.docker.internal also it gives an error but it gives a different error.

![image](https://user-images.githubusercontent.com/28344691/193536471-3870f625-d7ed-4856-8aac-4791d21c5c9c.png)

For server IP that is 192.168.18.138 it is working perfectly..

![image](https://user-images.githubusercontent.com/28344691/193536274-2705c829-0a0b-4f2a-a398-49850fa436b7.png)


in short from nocodb interface if i try to connect MySQL database with server IP then it works but localhost and host.docker.internal it gives error...

**Reply by patelbhavinoutlook** [reply]
One Update here after doing some more R&D...

tried to install MySQL NocoDB with the below standard command.. 

docker run -d --name nocodb-mysql \
-v "$(pwd)"/nocodb:/usr/app/data/ \
-p 8080:8080 \
-e NC_DB="mysql2://192.168.18.138:3306?u=testdb&p=password&d=testdb" \
-e NC_AUTH_JWT_SECRET="569a1821-0a93-45e8-87ab-eb857f20a010" \
nocodb/nocodb:latest

but it gave same error as below

Cannot GET /

Then removed -e NC_AUTH_JWT_SECRET="569a1821-0a93-45e8-87ab-eb857f20a010" \

And it worked.. i was able to access web interface at 8080..

**Comment by twicros** [comment]
So??? I have same problems. Cannot install Nocodb and Mysql in same server. Somebody have solution for that?
